File: app/src/main/java/com/kakeibo/ui/ItemStatusViewModel.java
//package com.kakeibo.ui;
//
//import android.app.Application;
//
//import androidx.lifecycle.AndroidViewModel;
//import androidx.lifecycle.LiveData;
//
//import com.kakeibo.SubApp;
//import com.kakeibo.data.DataRepository;
//import com.kakeibo.data.ItemStatus;
//
//import java.util.List;
//
//public class ItemStatusViewModel extends AndroidViewModel {
//
//    private DataRepository _repository;
//
//    private LiveData<List<ItemStatus>> _items;
//
//    public ItemStatusViewModel(Application application) {
//        super(application);
//        SubApp subApp = ((SubApp) application);
//        _repository = subApp.getRepository();
//        _items = _repository.getAll();
//    }
//
//    LiveData<List<ItemStatus>> getAll() { return _items; }
//
//    public void insert(ItemStatus itemStatus) { _repository.insertItemStatus(itemStatus); }
//}

File: app/src/main/java/com/kakeibo/ui/ItemStatusViewModel.kt
package com.kakeibo.ui

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.Transformations
import com.kakeibo.SubApp
import com.kakeibo.data.DataRepository
import com.kakeibo.data.ItemStatus

class ItemStatusViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: DataRepository = (application as SubApp).repository

    val all: LiveData<List<ItemStatus>> = repository.items
    val itemsByMonth: LiveData<Map<String, List<ItemStatus>>> = Transformations.map(all) { all ->
        all.groupBy ({ it.eventDate }, { it })
    }

    fun insert(itemStatus: ItemStatus) {
        repository.insertItem(itemStatus)
    }

    fun getItemsByMonth(year: String, month: String) {
        repository.getItemsByMonth(year, month)
    }

    fun calcDaySum(): String {

        return "asdf"
    }

    fun deleteAll() {
        repository.deleteAllItems()
    }
}